invenio_search/contrib/records.py

```diff
@@ -40,18 +42,40 @@
 @models_committed.connect
 def index_record_modification(sender, changes):
     """Example handler for indexing record metadata."""
+    index_default = current_app.config['SEARCH_INDEX_DEFAULT']
+    doc_type_default = current_app.config['SEARCH_DOC_TYPE_DEFAULT']
+    index_names = current_search.mappings.keys()
+
     for obj, change in changes:
         if isinstance(obj, RecordMetadata):
+            index, doc_type = index_default, doc_type_default
+            schema = obj.json.get('$schema')
+            if isinstance(schema, dict):
+                schema = schema.get('$ref')
+            if schema:
+                index, doc_type = schema_to_index(
+                    schema, index_names=index_names
+                )
```

Comment:
Just an idea. What about doing something like this:
In a file named `contrib/records_utils.py`
```python
def index_record_modification_factory(record_to_index_and_type):
  def index_record_modification(sender, changes):
      ...
      for obj, change in changes:
          if isinstance(obj, RecordMetadata):
              index, doc_type = record_to_index_and_type(record, index_default, doc_type_default)
    ...
    return index_record_modification

def record_to_index_and_type(record, index_default, doc_type_default):
    schema = record.json.get('$schema')
             if isinstance(schema, dict):
                 schema = schema.get('$ref')
             if schema:
                 index, doc_type = schema_to_index(
                     schema, index_names=index_names
                 )
```
In file `contrib/records.py`
```
models_committed.connect(index_record_modification_factory(record_to_index_and_type))
```

That way, if index or object type names can be decided otherwise and the indexer can still be used.